user_sync/rules.py

```diff
@@ -799,6 +809,13 @@ def update_umapi_users_for_connector(self, umapi_info, umapi_connector):
         umapi_info.set_umapi_users_loaded()
         return user_to_group_map
 
+    def get_umapi_user_in_groups(self,umapi_info, umapi_connector, groups):
```

Comment:
You can remove `self` from the argument list and make this a static method. (add the `@staticmethod` decorator)